nav2_controller/plugins/simple_goal_checker.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2017, Locus Robotics
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <memory>
+#include <string>
+#include "nav2_controller/plugins/simple_goal_checker.hpp"
+#include "pluginlib/class_list_macros.hpp"
+#include "angles/angles.h"
+#include "nav2_util/node_utils.hpp"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpedantic"
+#include "tf2/utils.h"
+#pragma GCC diagnostic pop
+
+namespace nav2_controller
+{
+
+SimpleGoalChecker::SimpleGoalChecker()
+: xy_goal_tolerance_(0.25),
+  yaw_goal_tolerance_(0.25),
+  stateful_(true),
+  check_xy_(true),
+  xy_goal_tolerance_sq_(0.0625)
+{
+}
+
+void SimpleGoalChecker::initialize(
+  const rclcpp_lifecycle::LifecycleNode::SharedPtr & nh,
+  const std::string & plugin_name)
+{
+  nav2_util::declare_parameter_if_not_declared(
+    nh,
+    plugin_name + ".xy_goal_tolerance", rclcpp::ParameterValue(0.25));
```

Comment:
If there are new / moved params, we need to:
- add them to the nav2_param.md file in `/docs/parameters`
- Add them to the navigation.ros.org migration guide
- Add them to the navigation.ros.org website for parameters, remove the old ones, and update the samples

Reply:
For all below as well for params that moved

Reply:
This looks like a different repo. Should I raise a different PR for this or it can be added to this one?

Reply:
... can you add a PR for another repo into this one? If so, I'm curious to see what that looks like - but otherwise another PR in that repo is the expected process (perhaps from lack of knowledge)

Reply:
Actually I was asking if that it possible. Doesn't look like. I have created a new PR for the doc changes.